solutions/security-enforced/variables.tf

```diff
@@ -0,0 +1,274 @@
+  }
+}
+
+variable "event_notifications_name" {
+  type        = string
+  description = "The name of the Event Notifications instance that is created by this solution. If a `prefix` input variable is specified, it is added to this name in the `<prefix>-value` format."
+  default     = "base-event-notifications"
+}
+
+variable "service_plan" {
+  type        = string
+  description = "The pricing plan of the Event Notifications instance. Possible values: `Lite`, `Standard`"
+  default     = "standard"
+  validation {
+    condition     = contains(["lite", "standard"], var.service_plan)
+    error_message = "The specified pricing plan is not available. The following plans are supported: `Lite`, `Standard`"
+  }
+
+}
+
+variable "event_notifications_resource_tags" {
+  type        = list(string)
+  description = "The list of tags to add to the Event Notifications instance."
+  default     = []
+}
+
+variable "event_notifications_access_tags" {
+  type        = list(string)
+  description = "A list of access tags to apply to the Event Notifications instance created by the module. For more information, see https://cloud.ibm.com/docs/account?topic=account-access-tags-tutorial."
+  default     = []
+
+  validation {
+    condition = alltrue([
+      for tag in var.event_notifications_access_tags : can(regex("[\\w\\-_\\.]+:[\\w\\-_\\.]+", tag)) && length(tag) <= 128
+    ])
+    error_message = "Tags must match the regular expression \"[\\w\\-_\\.]+:[\\w\\-_\\.]+\", see https://cloud.ibm.com/docs/account?topic=account-tag&interface=ui#limits for more details"
+  }
+}
+
+variable "existing_event_notifications_instance_crn" {
+  type        = string
+  description = "The CRN of existing event notification instance. If not supplied, a new instance is created."
+  default     = null
+}
+
+########################################################################################################################
+# KMS
+########################################################################################################################
+
+variable "existing_kms_instance_crn" {
+  type        = string
+  description = "The CRN of the KMS instance (Hyper Protect Crypto Services or Key Protect instance). If the KMS instance is in different account you must also provide a value for `ibmcloud_kms_api_key`."
+  default     = null
+}
+
+variable "existing_kms_root_key_crn" {
+  type        = string
+  description = "The key CRN of a root key, existing in the KMS instance passed in the `existing_kms_instance_crn` input, which will be used to encrypt the data. If no value passed, a new key will be created in the instance provided in the `existing_kms_instance_crn` input."
+  default     = null
+}
```

Comment:
You seem to be missing variable validation for the KMS inputs. For example: https://github.com/terraform-ibm-modules/terraform-ibm-scc/blob/6fbd4973c72855920c157848ab9eee0b7cce5439/solutions/security-enforced/variables.tf#L159-L181